add arm support docs (#558)
hashicorp/web-unified-docs: content/terraform-docs-agents/v1.22.x/docs/cloud-docs/agents/requirements.mdx

# Agent Requirements

> Ensure your system meets the following requirements before installing and configuring HCP Terraform Agents. Refer to [HCP Terraform Agents on Terraform Enterprise](/terraform/enterprise/admin/agents-on-tfe) for additional Terraform Enterprise requirements.

Reviewer: I think we should add this back 

~> Important: All agents in an agent pool should have the same operating system and hardware resources available. If the agents are using different resources, run performance can vary significantly between each agent in the pool.

Author: I added this language back into their respective environment and hardware sections. I did not use an alert box, which should be reserved for special cases.

> ## Architecture

Reviewer: ```suggestion
## Operating Systems and Architectures
```
I think it would be ok to have this be one section.

Reviewer: For the body of this section I think something like:

> [Agents](https://releases.hashicorp.com/tfc-agent/) currently support x86_64 bit and ARM64 Linux operating systems. You can also run the agent within Docker using our official [HCP Terraform Agent Docker container](https://hub.docker.com/r/hashicorp/tfc-agent).
> 
> The architecture of the agent pool is determined by the first agent that registers. For example, when the first agent to register to an empty pool is an ARM64 agent, subsequent agents that register must be ARM64 architecture agents. Attempting to register an ARM64 agent to an existing x86-only agent pool results in an error.
> 
> Refer to [Change the agent pool architecture](/terraform/cloud-docs/agents/agent-pools#change-the-agent-pool-architecture) for instructions on how to switch between architectures.

I don't think it is necessary to warn about provider incompatibility here because there is already a section explaining provider incompatibility errors and almost all providers have x64 and ARM64 builds, there are slightly fewer that have ARM64 builds so this is not something most customers need to be concerned about.

You can deploy agents to x86_64 and ARM64 Linux. You can also run the agent in Docker using our official [HCP Terraform Agent Docker container](https://hub.docker.com/r/hashicorp/tfc-agent).

The architecture of the agent pool is determined by the first agent that connects to Terraform. For example, when the first agent to register to an empty pool is an ARM64 agent, subsequent agents that register must be ARM64 agents. Attempting to register an ARM64 agent to an existing x86-only agent pool results in an error.

> Refer to [Change the agent pool architecture](/terraform/cloud-docs/agents/agent-pools#change-the-agent-pool-architecture) for instructions on how to switch between architectures.

Reviewer: The first paragraph of this section seems to be covered in the Operating Systems section below so I think it could be removed, or the agent pool details could be moved to a subsection of the Operating Systems section and then it could be removed. @radditude what do you think?

## Operating systems

> You can run the agent on Linux distributions for x86-64 or ARM64. You can also run the agent in a Docker container using the official [Terraform Cloud Agent image on Docker hub](https://hub.docker.com/r/hashicorp/tfc-agent).

Reviewer: ```suggestion
[Agents](https://releases.hashicorp.com/tfc-agent/) support x86_64 bit and ARM64 Linux operating systems. You can also run the agent within Docker using our official [HCP Terraform Agent Docker container](https://hub.docker.com/r/hashicorp/tfc-agent).
```

## Terraform versions

You can use the agent with the following versions of Terraform.

> - Agents on x86-64 architecture supports Terraform 0.12 and newer.
> - Agents on ARM64 architecture supports Terraform 0.13.5 and newer.

Reviewer: ```suggestion
- Terraform 0.12 and newer with x86-64 architecture agents.
- Terraform 0.13.5 and newer with ARM64 architecture agents.
```

# Agent Requirements

Ensure your system meets the following requirements before installing and configuring HCP Terraform Agents. Refer to [HCP Terraform Agents on Terraform Enterprise](/terraform/enterprise/admin/agents-on-tfe) for additional Terraform Enterprise requirements.

## Environment

You can deploy agents to x86_64 and ARM64 Linux. You can also run the agent in Docker using our official [HCP Terraform Agent Docker container](https://hub.docker.com/r/hashicorp/tfc-agent).

All agents in an agent pool should have the same operating system. If the agents use different resources, run performance can vary significantly between each agent in the pool.

> The architecture of the agent pool is determined by the first agent that connects to Terraform. For example, when the first agent to register to an empty pool is an ARM64 agent, subsequent agents that register must be ARM64 agents. Attempting to register an ARM64 agent to an existing x86-only agent pool results in an error.

Reviewer 2: A small language nitpick: is our current guideline to use "Terraform" when talking about both Terraform the CLI tool and HCP Terraform? Agent registration is entirely an HCP Terraform affair, so it may be more accurate to say:

> The architecture of the agent pool is determined by the first agent that connects to HCP Terraform.
